File: pset6/readability.py
import sys
from cs50 import get_string


def final_output(fs):
    if fs < 1:
        print("Before Grade 1")
    elif fs >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {fs}")


if __name__=='__main__':
    text = get_string("Text: ")
    words = text.split(" ")
    number_of_words = len(words)
    text2 = text.replace("?", ".")
    text3 = text2.replace("!", ".")
    sentences = text3.split(".")
    number_of_sentences = len(sentences)
    number_of_letters = 0
    for txt in words:
        number_of_letters+=len(txt)
    var1 = number_of_letters / number_of_words
    var2 = number_of_sentences / number_of_words
    score = 5.88 * var1  - 29.6 * var2 - 15.8
    final_score = round(score)
    final_output(final_score)

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string txt);
int count_words(string txt);
int count_sentences(string txt);
int readability_index(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);
    int index = readability_index(letters, words, sentences);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}


int count_letters(string txt)
{
    int letters = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
        if (isalpha(txt[i]))
        {
            letters+=1;
        }
    return letters;
}

int count_words(string txt)
{
    int words = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        if (txt[i] == '-' && isalpha(txt[i - 1])) {}

        else if (txt[i] == 39 && isalpha(txt[i - 1])) {}

        else if (!isalpha(txt[i]) && isalpha(txt[i - 1]))
        {
            words +=1;
        }

    }
    return words;
}

int count_sentences(string txt)
{
    int sentences = 0;
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        int character = txt[i];
        if (character == 46 || character == 33 || character == 63)
        {
            sentences +=1;
        }
    }
    return sentences;
}

int readability_index(int letters, int words, int sentences)
{
    float words_per_100 = words / 100.0;
    float L = letters / words_per_100;
    float S = sentences / words_per_100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}